starknet_py/hash/blake2s.py

```diff
@@ -0,0 +1,105 @@
+"""
+This module's Blake2s felt encoding and hashing logic is based on StarkWare's
+sequencer implementation:
+https://github.com/starkware-libs/sequencer/blob/main/crates/blake2s/src/lib.rs
```

Comment:
Let's use [parmalink](https://docs.github.com/en/repositories/working-with-files/using-files/getting-permanent-links-to-files) 😄